src/main/java/org/gaul/s3proxy/EncryptedBlobStore.java

```diff
@@ -0,0 +1,766 @@
+            InputStream isRaw = blob.getPayload().openStream();
+            Encryption encryption =
+                new Encryption(secretKey, isRaw, 1);
+            InputStream is = encryption.openStream();
+
+            // adjust the encrypted content length by
+            // adding the padding block size
+            long contentLength =
+                blob.getMetadata().getContentMetadata().getContentLength() +
+                    Constants.PADDING_BLOCK_SIZE;
+
+            return cipheredBlob(container, blob, is, contentLength, true);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Payload encryptPayload(Payload payload, int partNumber) {
+
+        try {
+            // open the streams and pass them through the encryption
+            InputStream isRaw = payload.openStream();
+            Encryption encryption =
+                new Encryption(secretKey, isRaw, partNumber);
+            InputStream is = encryption.openStream();
+
+            Payload cipheredPayload = Payloads.newInputStreamPayload(is);
+            cipheredPayload.setContentMetadata(payload.getContentMetadata());
+            cipheredPayload.setSensitive(payload.isSensitive());
+
+            // adjust the encrypted content length by
+            // adding the padding block size
+            long contentLength =
+                payload.getContentMetadata().getContentLength() +
+                    Constants.PADDING_BLOCK_SIZE;
+            cipheredPayload.getContentMetadata()
+                .setContentLength(contentLength);
+
+            return cipheredPayload;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Blob decryptBlob(Decryption decryption, String container,
+        Blob blob) {
+        try {
+            // handle blob does not exist
+            if (blob == null) {
+                return null;
+            }
+
+            // open the streams and pass them through the decryption
+            InputStream isRaw = blob.getPayload().openStream();
+            InputStream is = decryption.openStream(isRaw);
+
+            // adjust the content length if the blob is encrypted
+            long contentLength =
+                blob.getMetadata().getContentMetadata().getContentLength();
+            if (decryption.isEncrypted()) {
+                contentLength = decryption.getContentLength();
+            }
+
+            return cipheredBlob(container, blob, is, contentLength, false);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    // filter the list by showing the unencrypted blob size
+    private PageSet<? extends StorageMetadata> filteredList(
+        PageSet<? extends StorageMetadata> pageSet) {
+        ImmutableSet.Builder<StorageMetadata> builder = ImmutableSet.builder();
+        for (StorageMetadata sm : pageSet) {
+            if (sm instanceof BlobMetadata) {
```

Comment:
Do we need to also filter out the `.mpu/` blobs in GCS? I haven't gotten around to testing this against Azure/GCS, but seemed like they would show up?